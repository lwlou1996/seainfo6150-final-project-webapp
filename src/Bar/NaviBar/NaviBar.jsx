import React from 'react'

import styles from './NaviBar.module.css'

const NaviBar = () => {

    return(
        <div className={styles.navContainer}>
            <header className={styles.container}>
                <nav>
                    {/*<a href="/seainfo6150-final-project-webapp" className={styles.mainIcon}>POSub</a>*/}
                    <a href="/" className={styles.mainIcon}>POSub</a>
                </nav>

                <nav>
                    <ul className={styles.list}>
                        {/* these links should show you how to connect up a link to a specific route */}

                        <li className={styles.listItem}>
                            {/*<a href="/seainfo6150-final-project-webapp/about">About</a>*/}
                            <a href="/about">About</a>
                        </li>

                        <li className={styles.listItem}>
                            {/*<a href="/seainfo6150-final-project-webapp/category">Category</a>*/}
                            <a href="/category">Category</a>
                        </li>

                        {/*<li className={styles.listItem}>*/}
                        {/*    <a href="/seainfo6150-final-project-webapp/order">Order</a>*/}
                        {/*</li>*/}

                    </ul>
                </nav>
            </header>


        </div>



    );

}

export default NaviBar;