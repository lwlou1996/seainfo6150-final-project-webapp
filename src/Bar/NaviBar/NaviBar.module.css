@import url('https://fonts.googleapis.com/css2?family=Hind+Siliguri:wght@600&display=swap');


.container {
    display: flex;

    justify-content: space-between;

    margin: 10px 50px 5px 30px;

    /*border-bottom: gray solid 1px;*/
}

.mainIcon {
    font-family: 'Hind Siliguri', sans-serif;

    font-size: 2.0rem;

}

.list {
    display: flex;
    list-style-type: none;
}

.listItem {
    font-family: 'Hind Siliguri', sans-serif;
    margin-right: 1rem;

}

.section {
    margin-bottom: 2rem;
}

a:link {
    color: black;
    text-decoration: none;
}

a:visited {
    color: black;
    text-decoration: none;
}

a:hover {
    color: silver;
}