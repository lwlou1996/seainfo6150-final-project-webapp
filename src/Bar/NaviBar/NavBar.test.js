import React from 'react'
import {render} from "@testing-library/react";
import NaviBar from "./NaviBar";

describe('test navigation', () => {
    it('test navigation successfully',  () => {
        const {container} = render(<NaviBar />);
        expect(container).toMatchSnapshot();
    });
});