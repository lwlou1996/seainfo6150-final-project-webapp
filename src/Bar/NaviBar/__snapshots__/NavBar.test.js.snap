// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test navigation test navigation successfully 1`] = `
<div>
  <div
    class="navContainer"
  >
    <header
      class="container"
    >
      <nav>
        <a
          class="mainIcon"
          href="/"
        >
          PoSub
        </a>
      </nav>
      <nav>
        <ul
          class="list"
        >
          <li
            class="listItem"
          >
            <a
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li
            class="listItem"
          >
            <a
              href="/category"
            >
              Category
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>
</div>
`;
